app/lib/actions/katello/content_view_version/incremental_update.rb

```diff
@@ -63,8 +89,42 @@ def finalize
           history.save!
         end
 
+        # given a composite version, and a list of new components, calculate the list of all components for the new version
+        def calculate_components(old_version, new_components)
+          old_components = old_version.components.select do |component|
+            !old_version.components.map(&:content_view_id).include?(component.content_view_id)
+          end
+          old_components + new_components
+        end
+
         private
 
+        def validate_environments(to_environments, old_version)
+          unless (to_environments - old_version.environments).empty?
+            fail _("Content View Version %{id} not in all specified environments %{envs}") %
+                     {:id => old_version.id, :envs => (to_environments - old_version.environments).map(&:name).join(',')}
+          end
+        end
+
+        def validate_content(old_version, content, components)
+          if old_version.content_view.composite?
+            fail(_("Cannot specify content for composite views")) unless content.empty?
+            validate_components(old_version, components)
+          else
+            fail(_("Cannot specify components for non-composite views")) unless components.empty?
+          end
+        end
+
+        def validate_components(old_version, components)
+          old_component_content_view_ids = old_version.components.map(&:content_view_id)
+          components.each do  |cvv|
```

Comment:
Did you mean to have two spaces after `do` here? If not, I wonder if this may be something we could add to rubocop.